secret/secrets_manager_client.go

```diff
@@ -45,7 +53,25 @@ func NewBasicSecretsManagerClient(opts awsutil.ClientOptions) (*BasicSecretsMana
 
 // CreateSecret creates a new secret.
 func (c *BasicSecretsManagerClient) CreateSecret(ctx context.Context, in *secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error) {
-	return nil, errors.New("TODO: implement")
+	var out *secretsmanager.CreateSecretOutput
```

Comment:
See my above comment about `setup`. We also will have to call `setup` in `CreateSecret` before we can actually make the API call as a safety check against someone trying to construct a `BasicSecretsManagerClient` without using `NewBasicSecretsMangaerClient`. For example, this will panic:

```go
c := BasicSecretsManagerClient{}
c.CreateSecret(ctx, &secretsmanager.CreateSecretInput{})
```